api/agent.go

```diff
@@ -122,6 +122,7 @@ type AgentServiceConnectProxyConfig struct {
 }
 
 // AgentMember represents a cluster member known to the agent
+// AgentMemberStatus corresponds the Status integer to a string
```

Comment:
Shouldn't the `Status` field be of type `AgentMemberStatus` now? That change should be backwards compatible, because you can still assign an int to the field, even when the type has changed, because the type is defined as an `int`.

Reply:
Passed my basic testing, great suggestion! :)